tests/system/test_crud.py

```diff
@@ -1488,3 +1488,34 @@ class Container(ndb.Model):
 
     entity = entity.key.get()
     assert entity.child.foo == "bar"
+
+
+@pytest.mark.usefixtures("client_context")
+def test_local_structured_property_with_inheritance(dispose_of):
+    """Regression test for #523
+
+    https://github.com/googleapis/python-ndb/issues/523
+    """
+
+    class Base(ndb.Model):
+        pass
+
+    class SubKind(Base):
+        foo = ndb.StringProperty()
+
+    class Container(ndb.Model):
+        children = ndb.LocalStructuredProperty(Base, repeated=True)
+
+    entity = Container()
+
+    subkind = SubKind(foo="bar")
+    subkey = subkind.put()
+
+    entity.children.append(subkind)
+    key = entity.put()
+
+    dispose_of(key._key)
+    dispose_of(subkey._key)
+
+    entity = entity.key.get()
+    assert isinstance(entity.children[0], Base)
```

Comment:
Is it worth it to check `entity.children[0].foo`?

Reply:
This is a weird issue. The error came up with the latest release, but it turns out that using the base class as the property class never really worked properly. The child entity is of class Base after the put(), and it doesn't keep the "foo" attribute. In other words, this quick fix just leaves things as they were before for that use case, which means NOT working.

Avoiding the error thus fixes the issue for the user, though it doesn't fix the underlying problem. I took a good look at fixing that, but it's not a simple fix as far as I can tell. This was broken before anyway, so I thought I might fix the issue for our customer and then go back to it once this is merged.

Reply:
Ok, fair enough.